src/d/d_tree.cpp

```diff
@@ -32,33 +33,52 @@ void dTree_data_c::animation(int) {
 }
 
 /* 8007945C-8007946C       .text newData__12dTree_room_cFP12dTree_data_c */
-void dTree_room_c::newData(dTree_data_c*) {
-    /* Nonmatching */
+void dTree_room_c::newData(dTree_data_c* data) {
+    data->mpNext = this->mpHead;
+    this->mpHead = data;
+    return;
 }
 
 /* 8007946C-800794D0       .text deleteData__12dTree_room_cFv */
 void dTree_room_c::deleteData() {
-    /* Nonmatching */
+    while (this->mpHead != (dTree_data_c*)0x0) {
+        this->mpHead->field_0x000 = 0;
+        mDoAud_seDeleteObject(&this->mpHead->mPos);
+        this->mpHead = this->mpHead->mpNext;
+    }
+    return;
 }
 
 /* 800794D0-800795E8       .text __ct__14dTree_packet_cFv */
 dTree_packet_c::dTree_packet_c() {
-    /* Nonmatching */
+    dTree_data_c* pdVar1;
+    dTree_anm_c* pdVar2;
+    short sVar3;
+    int iVar4;
+
+    this->field_0x0010 = 0;
+    this->field_0x6734 = 0;
+    this->field_0x6736 = 0;
+    pdVar1 = this->mData;
+    iVar4 = 0x40;
+    return;
```

Comment:
Please leave the Nonmatching in if it doesn't match. This doesn't even look close. The loop below is probably something like:

```cpp
for (s32 i = 0; i < (s32)ARRAY_SIZE(mData); i++)
    mData[i].field_0x000 = 0;
```